archunit/src/main/java/com/tngtech/archunit/core/domain/JavaClass.java

```diff
@@ -1048,55 +1072,71 @@ public JavaConstructor getConstructor(String... parameters) {
      */
     @PublicAPI(usage = ACCESS)
     public Set<JavaField> getFieldsWithTypeOfSelf() {
-        return javaClassDependencies.getFieldsWithTypeOfClass();
+        return reverseDependencies.getFieldsWithTypeOf(this);
     }
 
     /**
      * @return Methods of all imported classes that have a parameter type of this class.
      */
     @PublicAPI(usage = ACCESS)
     public Set<JavaMethod> getMethodsWithParameterTypeOfSelf() {
-        return javaClassDependencies.getMethodsWithParameterTypeOfClass();
+        return reverseDependencies.getMethodsWithParameterTypeOf(this);
     }
 
     /**
      * @return Methods of all imported classes that have a return type of this class.
      */
     @PublicAPI(usage = ACCESS)
     public Set<JavaMethod> getMethodsWithReturnTypeOfSelf() {
-        return javaClassDependencies.getMethodsWithReturnTypeOfClass();
+        return reverseDependencies.getMethodsWithReturnTypeOf(this);
     }
 
     /**
      * @return {@link ThrowsDeclaration ThrowsDeclarations} of all imported classes that have the type of this class.
      */
     @PublicAPI(usage = ACCESS)
     public Set<ThrowsDeclaration<JavaMethod>> getMethodThrowsDeclarationsWithTypeOfSelf() {
-        return javaClassDependencies.getMethodThrowsDeclarationsWithTypeOfClass();
+        return reverseDependencies.getMethodThrowsDeclarationsWithTypeOf(this);
     }
 
     /**
      * @return Constructors of all imported classes that have a parameter type of this class.
      */
     @PublicAPI(usage = ACCESS)
     public Set<JavaConstructor> getConstructorsWithParameterTypeOfSelf() {
-        return javaClassDependencies.getConstructorsWithParameterTypeOfClass();
+        return reverseDependencies.getConstructorsWithParameterTypeOf(this);
     }
 
     /**
      * @return {@link ThrowsDeclaration ThrowsDeclarations} of all imported classes that have the type of this class.
      */
     @PublicAPI(usage = ACCESS)
     public Set<ThrowsDeclaration<JavaConstructor>> getConstructorsWithThrowsDeclarationTypeOfSelf() {
-        return javaClassDependencies.getConstructorsWithThrowsDeclarationTypeOfClass();
+        return reverseDependencies.getConstructorsWithThrowsDeclarationTypeOf(this);
     }
 
     /**
      * @return All imported {@link JavaAnnotation JavaAnnotations} that have the annotation type of this class.
      */
     @PublicAPI(usage = ACCESS)
     public Set<JavaAnnotation<?>> getAnnotationsWithTypeOfSelf() {
-        return javaClassDependencies.getAnnotationsWithTypeOfClass();
+        return reverseDependencies.getAnnotationsWithTypeOf(this);
+    }
+
+    /**
+     * @return All imported {@link JavaAnnotation JavaAnnotations} that have a parameter with type of this class.
+     */
+    @PublicAPI(usage = ACCESS)
+    public Set<JavaAnnotation<?>> getAnnotationsWithParameterTypeOfSelf() {
+        return reverseDependencies.getAnnotationsWithParameterTypeOf(this);
```

Comment:
According to the test coverage, this can – as well as `getInstanceofChecksWithTypeOfSelf` – _just be simplified_ to:
```
return Collections.emptySet();
```
Maybe we should have some tests for those new methods...? 😜

Reply:
Ah yes, now I remember that I wanted to add some additional test :see_no_evil: In particular, because for the final state this method is not used anymore to create the reverse dependencies :wink:

Reply:
Okay, I added tests for the two new public API methods :smiley: